payments-core/src/main/java/com/stripe/android/ConfirmStripeIntentParamsFactory.kt

```diff
@@ -17,7 +17,10 @@ import com.stripe.android.model.SetupIntent
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
 sealed class ConfirmStripeIntentParamsFactory<out T : ConfirmStripeIntentParams> {
 
-    abstract fun create(paymentMethod: PaymentMethod): T
+    abstract fun create(
+        paymentMethod: PaymentMethod,
+        requiresSaveOnConfirmation: Boolean = false,
```

Comment:
Do we have tests for passing the right value into this?

Reply:
Added some network tests for ensuring the value is properly passed in PM and passthrough mode.

Reply:
Is it possible to have unit tests as well?

Reply:
Added unit tests in `PaymentSheetViewModel`.